pkgs/stdenv/generic/check-meta.nix

```diff
@@ -659,6 +663,9 @@ let
       # or nix-env complains: https://github.com/NixOS/nix/blob/2.18.8/src/nix-env/nix-env.cc#L963
       ${if maintainersPosition == null then null else "maintainersPosition"} = maintainersPosition;
       ${if teamsPosition == null then null else "teamsPosition"} = teamsPosition;
+
+      # Default to an application since most packages are applications
+      type = [ packageTypes.application ];
```

Comment:
Thinking more about it, I don't think this should default to anything.

This should instead leave it to the consuming software to decide what to do with `null` (or otherwise) entries.

I actually doubt that most packages are applications.

Think about the implications of the following attribute path: `pkgs.hello.src.meta`

Reply:
After this PR, a treewide can happen to properly set things. `pkgs.hello.src.meta.type` would likely be `data` since it's data, it's source code. Though, if it is a tarball then `file` should be used. It's just a lot of things are applications so it's set to application to make it so it doesn't have to be put everywhere.

Reply:
Hmm, makes sense.